src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -318,14 +319,15 @@ public void Dispose()
             RuntimeCapabilities capabilities,
             bool isExecutable,
             NonNullable<string> processorArchitecture,
-            References externals = null,
-            IEnumerable<ReaderWriterLockSlim> dependentLocks = null)
+            References? externals = null,
+            IEnumerable<ReaderWriterLockSlim>? dependentLocks = null)
         {
+            externals ??= References.Empty;
+            dependentLocks ??= Enumerable.Empty<ReaderWriterLockSlim>();
+
             this.syncRoot = new ReaderWriterLockSlim(LockRecursionPolicy.SupportsRecursion);
-            this.dependentLocks = dependentLocks == null
-                ? new HashSet<ReaderWriterLockSlim>()
-                : new HashSet<ReaderWriterLockSlim>(dependentLocks);
-            if (dependentLocks?.Contains(null) ?? false)
+            this.dependentLocks = new HashSet<ReaderWriterLockSlim>(dependentLocks);
+            if (dependentLocks.Contains(null!))
```

Comment:
What does the `!` do here? Is that new syntax for the nullable references?

Reply:
I guess this is just the null-forgiveness operator applied to a null value, because now references can't be null by default.